src/main/java/com/kelinci/mainapp/api/UserController.java

```diff
@@ -9,24 +9,18 @@
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Objects;
+import java.util.stream.Collectors;
 
 @RestController
 public class UserController {
-    private OurUser registeredUser;
-    private OurUser confirmedUser;
+    private OurUser ourUser;
 
     private List<OurUser> listOfUsers = new ArrayList<>();
 
-    @GetMapping(value = "/registered/lastregistereduser")
+    @GetMapping(value = "/registered/lastuser")
```

Comment:
Tak nawiasem mówiąc: wartości url endpointa nie można zmieniać sobie tak, o! Ba! Nie wolno w ogóle zmieniać! 
Raz wystawiony publicznie endpoint powinien być już dostępny. Pytanie do Ciebie: Dlaczego?

Reply:
ooo - haha, a ja sobie tak dowolnie :) ok, czyli jak już wystawiamy, to nie modyfikujemy. Ale nie wiem czemu - może jak już jest wystawiony to pozostaje aktywny? Czyli w sumie to jeśli tych endpointów nie modyfikujemy potem to trzeba dobrze przemyśleć strukturę aplikacji na samym początku - a co jeśli nam się koncepcja zmieni?

Reply:
Dokładnie tak jak powiedziałeś.
Jak już raz wystawimy endpointy to ktoś może się na nie już zapiąć (czyli zacznie ich używać). Więc my nie możemy robić zmiany, która nie jest wstecz kompatybilna (a zmiana endpointu takowa nie jest :P )
Endpoint trzeba przemyśleć + można robić zmiany kompatybilne wstecz (np dodawać nowe pola do requestu / response)
Jak koncepcja się zmieni to w ostateczności robi się nowy endpoint